lib/ansible/plugins/lookup/aws_secret.py

```diff
@@ -0,0 +1,135 @@
+# Copyright: (c) 2018, Aaron Smith <ajsmith10381@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+lookup: aws_secret
+author:
+  - Aaron Smith <ajsmith10381@gmail.com>
+version_added: "2.7"
+requirements:
+  - boto3
+  - botocore>=1.10.0
+extends_documentation_fragment:
+  - aws_credentials
+short_description: Look up secrets stored in AWS Secrets Manager.
+description:
+  - Look up secrets stored in AWS Secrets Manager provided the caller
+    has the appropriate permissions to read the secret.
+  - Lookup is based on the secret's `Name` value.
+  - Optional parameters can be passed into this lookup; `version_id` and `version_stage`
+options:
+  _terms:
+    description: Name of the secret to look up in AWS Secrets Manager.
+    required: True
+  version_id:
+    description: Version of the secret(s).
+    required: False
+  version_stage:
+    description: Stage of the secret version.
+    required: False
+  join:
+    description:
+        - Join two or more entries to form an extended secret.
+        - This is useful for overcoming the 4096 character limit imposed by AWS.
+    type: boolean
+    default: false
+"""
+
+EXAMPLES = r"""
+ - name: Create RDS instance with aws_secret lookup for password param
+   rds:
+     command: create
+     instance_name: app-db
+     db_engine: MySQL
+     size: 10
+     instance_type: db.m1.small
+     username: dbadmin
+     password: "{{ lookup('aws_secret', 'DbSecret') }}"
+     tags:
+       Environment: staging
+"""
+
+RETURN = r"""
+_raw:
+  description:
+    Returns the value of the secret stored in AWS Secrets Manager.
+"""
+
+from ansible.errors import AnsibleError
+
+try:
+    import boto3
+    import botocore
+except ImportError:
+    raise AnsibleError("The lookup aws_secret requires boto3 and botocore.")
+
+from ansible.plugins import AnsiblePlugin
+from ansible.plugins.lookup import LookupBase
+from ansible.module_utils.ec2 import boto3_conn, get_aws_connection_info
+from ansible.module_utils._text import to_native
+from ansible.module_utils.six import string_types
+import os
+
+
+def _boto3_conn(region, credentials):
+    boto_profile = credentials.pop('aws_profile', None)
+
+    try:
+        connection = boto3.session.Session(profile_name=boto_profile).client('secretsmanager', region, **credentials)
+    except (botocore.exceptions.ProfileNotFound, botocore.exceptions.PartialCredentialsError) as e:
+        if boto_profile:
+            try:
+                connection = boto3.session.Session(profile_name=boto_profile).client('secretsmanager', region)
```

Comment:
If this fails, it would be good to be able to fall back to instance role credentials if they exist. Here's a commit fixing that for another plugin for reference if it helps https://github.com/ansible/ansible/commit/d5a5e37fcf39de757ebe66bb239d9b9a5837b9ab#diff-fde53b7bca9c303ec46c0a71219bab47

Reply:
ok, I pushed a fix for your comments except this one.

This part of the code looks to be a copy of aws_ssm plugin. shouldn't the credentials retrieval be factorized somewhere in aws common code ?